src/Utils/FileSystem.php

```diff
@@ -84,6 +84,14 @@ public static function delete($path)
 			if (!@rmdir($path)) { // @ is escalated to exception
 				throw new Nette\IOException("Unable to delete directory '$path'.");
 			}
+		} elseif ($path instanceof \Nette\Utils\Finder) {
+			/** @var \SplFileInfo $item */
+			foreach ($path as $item) {
+				static::delete($item->getRealPath());
+			}
+			if (!@rmdir($item->getPath())) { // @ is escalated to exception
```

Comment:
Because I don't know if I will erase whole directory or not. There is missing check for it.